dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/program/DefaultProgramIndicatorService.java

```diff
@@ -485,8 +493,17 @@ public String expressionIsValid( String expression )
             return expr;
         }
 
-        if ( !ExpressionUtils.isValid( expr, null ) )
+        // Catch any exceptions from interpreting the expression, incase the expression is invalid.
+        try
+        {
+            if ( !ExpressionUtils.isValid( expr, null ) )

```

Comment:
I would suggest we move the exception catching into the isValid method. That way we encapsulate the check rather than than spreading it throughout the code.